src/plugins/intel_gna/src/transformations/ts_split.cpp

```diff
@@ -0,0 +1,135 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/ts_split.hpp"
+
+#include <openvino/cc/ngraph/itt.hpp>
+
+#include "backend/gna_limitations.hpp"
+#include "common/graph_utils.hpp"
+#include "openvino/core/rt_info.hpp"
+#include "openvino/op/util/op_types.hpp"
+#include "openvino/opsets/opset10.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "transformations/rt_info/transpose_sinking_attr.hpp"
+#include "transformations/utils/transformation_helper.hpp"
+
+using namespace ov;
+using namespace ov::opset10;
+using namespace ov::pass::pattern;
+using namespace ov::intel_gna::pass;
+using namespace ov::intel_gna::pass::helper;
+using namespace ov::intel_gna::limitations;
+using namespace ov::intel_gna::graph_utils;
+
+namespace {
+bool is_sinked(const Output<Node>& output) {
+    auto split_node = output.get_node_shared_ptr();
+    for (size_t output_idx = 0; output_idx < split_node->get_output_size(); ++output_idx) {
+        for (auto& input : split_node->get_output_target_inputs(output_idx)) {
+            auto transpose = ov::as_type_ptr<Transpose>(input.get_node()->shared_from_this());
+            if (transpose && !Limitations::is_transpose_supported(transpose))
+                return true;
+        }
+    }
+    return false;
+}
+}  // namespace
+
+TSSplitBackward::TSSplitBackward() {
+    MATCHER_SCOPE(TSSplitBackward);
+
+    auto split_node_label = wrap_type<Split>(is_sinked);
+
+    matcher_pass_callback matcher_pass_callback = [=](Matcher& m) {
+        const auto& pattern_to_output = m.get_pattern_value_map();
+
+        auto& split_node_label_output = pattern_to_output.at(split_node_label);
+        auto split_node = as_type_ptr<Split>(split_node_label_output.get_node_shared_ptr());
+
+        std::vector<AxisVector> gather_indices_vecs;
+        for (size_t output_idx = 0; output_idx < split_node->get_output_size(); ++output_idx) {
+            for (auto& input : split_node->get_output_target_inputs(output_idx)) {
+                auto transpose = ov::as_type_ptr<Transpose>(input.get_node()->shared_from_this());
+
+                if (transpose && !Limitations::is_transpose_supported(transpose)) {
+                    auto transpose_const = ov::as_type_ptr<Constant>(transpose->get_input_node_shared_ptr(1));
+                    if (!transpose_const)
+                        return false;
+                    auto gather_indices_value =
+                        make_gather_indices_from_transpose_axes(transpose->get_input_shape(0),
+                                                                transpose_const->get_axis_vector_val());
+                    gather_indices_vecs.push_back(gather_indices_value);
+                } else {
+                    const Shape& input_shape = input.get_shape();
```

Comment:
it incorrectly generates indexes for inputs without Transpose, please take the implementation from layout transformations branch

Reply:
Discussed on the meeting. We will create separate PR for it